services/common/src/main/java/org/finra/gatekeeper/common/authfilter/parser/IGatekeeperUserProfile.java

```diff
@@ -18,7 +18,13 @@
 package org.finra.gatekeeper.common.authfilter.parser;
 
 import java.security.Principal;
+import java.util.Set;
 
 public interface IGatekeeperUserProfile extends Principal {
+    public String getUserId();
+    public String getName();
+    public String getEmail();
```

Comment:
I would define a new interface that extends this one and have the additional methods specified in there.